### ISSUE TITLE
Request: [clang64] LTO

### ISSUE THREAD

**Issue by esator** [jb-alvarado/media-autobuild_suite#2669]
After recent clang changes, now LTO is possible for clang with `-flto=thin` via custom_build_options and `--enable-lto=thin` for ffmpeg
Also x264 needs `--enable-lto` since it has linking errors because by default forces `-mstack-alignment=64`, but for ffmpeg and other libs it's `-mstack-alignment=16`
It would be nice to have some option to enable lto for clang, nowadays lto is quite common and compatible, also `-flto=thin` is just a bit slower than normal compilation, also some libs and tools may require individual flags for lto (like `-DSVT_AV1_LTO=ON` for svt-av1, `-DENABLE_LTO` for x265, etc)
At least as an experimental and unsupported option, because it might require more maintenance and have less compatibility

**Comment by gitoss** [comment]
> > I don't know how much effect it would have with llvm though.
> 
> see [llvm/llvm-project#91862](https://github.com/llvm/llvm-project/pull/91862)

Right, so it's probably good to limit lto to core encoder libs/binaries that would gain speed.

I just compiled x265 and svt-av1 with lto by adding the -C and -A args to the .sh, seems to have worked fine and didn't take ages.
